src/libxtp/bulkesp.cc

```diff
@@ -0,0 +1,768 @@
+                Grid _grid(true, false, false); //create polarsites, so we can output grid to .cube file
+                //            _grid.setAtomlist(&m->atoms);
+                //            _grid.setupCHELPgrid();
+                _grid.setCubegrid(true);
+                if (periodic) {
+                    //_grid.setPadding(0.0);
+                    _grid.setPeriodicity(boxLen);
+                }
+                
+//                //let's just consider the first Oxygen atom as a molecule
+//                m->atoms= vector<ctp::QMAtom*>(m->atoms.begin()+0, m->atoms.begin()+2);
+//                m->atomIndeces= vector<unsigned>(m->atomIndeces.begin()+0, m->atomIndeces.begin()+2);
+
+                //test: set inner cutoff to 0 and calculate all potentials near nuclei
+//                _grid.setCutoffs(20, 0.001); //between 1.5 and 3 A, as that is the region where water-water interactions take place
+//                _grid.setSpacing(0.3); //defaults to 0.3 A
+                _grid.setAtomlist(&m->atoms);
+//                _grid.setAtomlist(&_atomlist);
+                _grid.setupgrid();
+                _grid.setupCHELPgrid();
+                CTP_LOG(ctp::logDEBUG, *_log) << ctp::TimeStamp() << " Done setting up CHELPG grid with " << _grid.getsize() << " points " << flush;
+
+                //calculate the ESP
+                //ub::vector<double> ESP=ComputeESP(m->atoms, _m_dmat, _m_ovmat, _m_basis, bs, gridsize, _grid);
+                double netcharge = 0.0;
+                ub::vector<double> ESP = ComputeESP(_atomlist, m->atoms, m->atomIndeces,
+                        _global_dmat, _basis, bs, gridsize, _grid, netcharge);
+                
+//                std::vector< unsigned > allIndeces;
+//                for (vector<ctp::QMAtom*>::iterator i = _atomlist.begin(); i != _atomlist.end(); ++i) {
+//                    allIndeces.push_back(i - _atomlist.begin());
+//                }
+//                ub::vector<double> ESP = ComputeESP(_atomlist, _atomlist, allIndeces,
+//                        _global_dmat, _basis, bs, gridsize, _grid, netcharge);
+
+                //store the potential in apolarsites
+                for (unsigned int i = 0; i < _grid.getsize(); i++) {
+                    //                ub::vector<double> point = _grid.getGrid()[i];
+                    //                APolarSite* site = _grid.Sites()[i];
+                    //                site->setPhi(ESP(i), 0.0);
+                    _grid.Sites()[i]->setPhi(ESP(i), 0.0);
+                }
+                
+                system_netcharge+=netcharge;
+
+
+                //and save it to a .cube file
+                std::ostringstream fn;
+                if (periodic && _grid.getCubegrid()) {
+                    fn.str(std::string());
+                    fn << "BulkEsp_" << m - mols.begin() << ".cube";
+                    _grid.printgridtoCubefile(fn.str());
+                }
+
+                //output
```

Comment:
make this its own object. Do you have to create a new file format? Why not use cube and set entries to zero?

Reply:
Good idea.

Reply:
Actually the code already does both. The custom format is just a lot faster to process with python for the rest of my research. But for the merge it should probably be removed.